.github/scripts/otp-compliance.es

```diff
@@ -653,74 +806,636 @@ curated_path_license(_Name, Path, [#{<<"path">> := Path}=Cur | _Curations]) ->
+                             nomatch ->
+                                 false;
+                             _ ->
+                                 true
+                         end
+                 end, Files).
+
+test_file(#{sbom_file := SbomFile}) ->
+    Sbom = decode(SbomFile),
+    test_generator(Sbom).
+
+test_generator(Sbom) ->
+    ok = project_generator(Sbom),
+    ok = package_generator(Sbom),
+    ok.
+
+project_generator(Sbom) ->
+    ok = test_project_name(Sbom),
+    ok = test_name(Sbom),
+    ok = test_creators_tooling(Sbom),
+    ok = test_spdx_version(Sbom),
+    ok.
+
+package_generator(Sbom) ->
+    ok = test_minimum_apps(Sbom),
+    ok = test_copyright_not_empty(Sbom),
+    ok = test_filesAnalised(Sbom),
+    ok = test_hasFiles_not_empty(Sbom),
+    ok = test_homepage(Sbom),
+    ok = test_licenseConcluded_exists(Sbom),
+    ok = test_licenseDeclared_exists(Sbom),
+    ok = test_licenseInfoFromFiles_not_empty(Sbom),
+    ok = test_package_names(Sbom),
+    ok = test_package_ids(Sbom),
+    ok = test_verificationCode(Sbom),
+    ok = test_supplier_Ericsson(Sbom),
+    ok = test_originator_Ericsson(Sbom),
+    ok = test_versionInfo_not_empty(Sbom),
+    ok = test_package_hasFiles(Sbom),
+    ok = test_project_purl(Sbom),
+    ok = test_packages_purl(Sbom),
+    ok = test_download_location(Sbom),
+    ok = test_package_relations(Sbom),
+    %% ok = test_doc_relation(Sbom),
+    ok = test_vendor_packages(Sbom),
+    ok.
+
+test_project_name(#{~"documentDescribes" := [ProjectName]}=_Sbom) ->
+    ?spdxref_project_name = ProjectName,
+    ok.
+
+test_name(#{~"name" := Name}=_Sbom) ->
+    ?spdx_project_name = Name,
+    ok.
+
+test_creators_tooling(#{~"creationInfo" := #{~"creators" := Creators}}=_Sbom) ->
+    true = lists:any(fun (Name) ->
+                             case string:prefix(Name, ?spdx_creators_tooling) of
+                                 nomatch -> false;
+                                 _ -> true
+                             end
+                     end, Creators),
+    ok.
+
+test_spdx_version(#{~"spdxVersion" := Version}=_Sbom) ->
+    ?spdx_version = Version,
+    ok.
+
+test_minimum_apps(#{~"documentDescribes" := [ProjectName], ~"packages" := Packages}=_Sbom) ->
+    _ = lists:foreach(fun (X) -> application:load(erlang:binary_to_atom(X)) end, minimum_otp_apps()),
+    [#{~"name" := Project}] = lists:filter(fun (#{~"SPDXID" := Id}) -> Id == ProjectName end, Packages),
+    TestPackageNames = [Project | minimum_otp_apps() ++ root_vendor_packages()],
+    SPDXIds = lists:map(fun (#{~"name" := SPDXId}) -> SPDXId end, Packages),
+    try
+        %% test know packages are captured
+        true = [] == TestPackageNames -- SPDXIds
+    catch
+        _E:_S:_ ->
+            io:format("Minimum apps not captured.~n~p distinct from ~p~n", [TestPackageNames -- SPDXIds, SPDXIds -- TestPackageNames])
+            %% error(?FUNCTION_NAME)
+    end,
+    ok.
+
+minimum_otp_apps() ->
```

Comment:
What about the "system" application? The one that contains all system documentation, do we need a package for that?

Reply:
no, I think system only has docs and lives inside the OTP project package.
@rickard-green correct me if I misunderstood how to package things :)